src/map/unit.c

```diff
@@ -238,226 +238,224 @@ static int unit_step_timer(int tid, int64 tick, int id, intptr_t data)
 
-	//Monsters can only leave icewalls to the west and south
-	//But if movement fails more than icewall_walk_block times, they can ignore this rule
-	if (md && md->walktoxy_fail_count < icewall_walk_block && map->getcell(bl->m, bl, x, y, CELL_CHKICEWALL) && (dx > 0 || dy > 0)) {
-		//Needs to be done here so that rudeattack skills are invoked
+	// Monsters can only leave icewalls to the west and south
+	// But if movement fails more than icewall_walk_block times, they can ignore this rule
+	if (md != NULL && md->walktoxy_fail_count < icewall_walk_block && map->getcell(bl->m, bl, x, y, CELL_CHKICEWALL) && (dx > 0 || dy > 0)) {
+		// Needs to be done here so that rudeattack skills are invoked
 		md->walktoxy_fail_count++;
 		clif->fixpos(bl);
-		//Monsters in this situation first use a chase skill, then unlock target and then use an idle skill
-		if (!(++ud->walk_count%WALK_SKILL_INTERVAL))
+		// Monsters in this situation first use a chase skill, then unlock target and then use an idle skill
+		if ((++ud->walk_count % WALK_SKILL_INTERVAL) != 0)
 			mob->skill_use(md, tick, -1);
 		mob->unlocktarget(md, tick);
-		if (!(++ud->walk_count%WALK_SKILL_INTERVAL))
+		if ((++ud->walk_count % WALK_SKILL_INTERVAL) != 0)
 			mob->skill_use(md, tick, -1);
-		return 0;
+		return 1;
 	}
 
 	//Refresh view for all those we lose sight
-	map->foreachinmovearea(clif->outsight, bl, AREA_SIZE, dx, dy, sd?BL_ALL:BL_PC, bl);
+	map->foreachinmovearea(clif->outsight, bl, AREA_SIZE, dx, dy, (sd != NULL ? BL_ALL : BL_PC), bl);
 
 	x += dx;
 	y += dy;
 	map->moveblock(bl, x, y, tick);
-	ud->walk_count++; //walked cell counter, to be used for walk-triggered skills. [Skotlex]
+	ud->walk_count++; // walked cell counter, to be used for walk-triggered skills. [Skotlex]
 	status_change_end(bl, SC_ROLLINGCUTTER, INVALID_TIMER); //If you move, you lose your counters. [malufett]
 
 	if (bl->x != x || bl->y != y || ud->walktimer != INVALID_TIMER)
-		return 0; //map->moveblock has altered the object beyond what we expected (moved/warped it)
+		return 1; // map->moveblock has altered the object beyond what we expected (moved/warped it)
 
 	ud->walktimer = -2; // arbitrary non-INVALID_TIMER value to make the clif code send walking packets
-	map->foreachinmovearea(clif->insight, bl, AREA_SIZE, -dx, -dy, sd?BL_ALL:BL_PC, bl);
+	map->foreachinmovearea(clif->insight, bl, AREA_SIZE, -dx, -dy, (sd != NULL ? BL_ALL : BL_PC), bl);
 	ud->walktimer = INVALID_TIMER;
 
-	if(sd) {
-		if( sd->touching_id )
-			npc->touchnext_areanpc(sd,false);
+	if (sd != NULL) {
+		if (sd->touching_id != 0)
+			npc->touchnext_areanpc(sd, false);
 		if (map->getcell(bl->m, bl, x, y, CELL_CHKNPC)) {
-			npc->touch_areanpc(sd,bl->m,x,y);
-			if (bl->prev == NULL) //Script could have warped char, abort remaining of the function.
-				return 0;
-		} else
+			npc->touch_areanpc(sd, bl->m, x, y);
+			nullpo_retr(1, bl->prev); //Script could have warped char, abort remaining of the function.
```

Comment:
If it's something that could happen (the script warping the character), shouldn't it stay as an if/return instead of nullpo_retr?

Reply:
Yes, you're right.

Reply:
Done.